pinot-common/src/main/java/org/apache/pinot/common/function/FunctionRegistry.java

```diff
@@ -140,8 +133,30 @@ public static boolean containsFunction(String functionName) {
    */
   @Nullable
   public static FunctionInfo getFunctionInfo(String functionName, int numParameters) {
-    Map<Integer, FunctionInfo> functionInfoMap = FUNCTION_INFO_MAP.get(canonicalize(functionName));
-    return functionInfoMap != null ? functionInfoMap.get(numParameters) : null;
+    Collection<Map.Entry<String, Function>> allFunctions = FUNCTION_MAP.range(canonicalize(functionName), true);
+    if (allFunctions.isEmpty()) {
+      return null;
+    }
+
+    List<Function> matchingFunctions = allFunctions
+        .stream()
+        .map(Map.Entry::getValue)
+        .filter(fun -> fun.getParameters().size() == numParameters)
+        .collect(Collectors.toList());
```

Comment:
if we can we should avoid doing stream compute on every get function info.

Reply:
getting function info happens once (per function) per query, right? not on every invocation of it? I'll remove it just in case, but I'd be shocked if things that happen once per query were a bottleneck

Reply:
yeah technically it should be fine as you say - but it is not once per query, it could be `O(#_of_column * #_of_function_usage)`

I think your goal is to remove the parameter-number-based mapper altogether. can we 
- keep the static final object so the lookup is fast. 
- remove all access so that no one can directly modify this function map?

once we upgrade all code lookup to be type-based we can remove them all together.